adlfs/spec.py

```diff
@@ -960,8 +952,6 @@ async def _find(self, path, withdirs=False, prefix="", with_parent=False, **kwar
                 files[file["name"]] = file
 
         if withdirs:
-            if not with_parent:
-                dirs.pop(target_path, None)
```

Comment:
with_parent is not present in fsspec/gcsfs/s3fs nor it is explicitly tested anywhere. It was getting in the way, so I've removed it as well.